plugins/Initial/src/Initial.ts

```
@@ -25,15 +26,33 @@ class InitialPluginInstance {
     }
     getInitial = (path: Path) => {
         let result = this.initialState;
-        path.forEach(p => {
+        path.forEach((p) => {
             result = result && result[p];
         });
         return result;
-    }
+    };
     getModified = (l: StateMethods<StateValueAtPath>): boolean => {
-        l.attach(Downgraded)
-        return !isEqual(l.value, this.getInitial(l.path))
-    }
+        l.attach(Downgraded);
+        return !isEqual(l.value, this.getInitial(l.path));
+    };
+    set = (path: Path, s: StateValueAtPath) => {
+        let result = this.initialState;
+        if (path.length > 0) {
+            for (let i = 0; i < path.length - 1; i++) {
+                result = result && result[path[i]];
```

I think it should throw if result is null / undefined here.

---

otherwise, it will modify parent state unexpectedly